lighthouse-core/audits/installable-manifest.js

```diff
@@ -51,59 +125,108 @@ class InstallableManifest extends MultiCheckAudit {
   }
 
   /**
-   * @param {LH.Artifacts.ManifestValues} manifestValues
-   * @return {Array<string>}
+   * @param {LH.Artifacts} artifacts
+   * @return {{i18nErrors: Array<LH.IcuMessage | string>; warnings: Array<LH.IcuMessage>}}
    */
-  static assessManifest(manifestValues) {
-    if (manifestValues.isParseFailure && manifestValues.parseFailureReason) {
-      return [manifestValues.parseFailureReason];
+  static getInstallabilityErrors(artifacts) {
+    const installabilityErrors = artifacts.InstallabilityErrors.errors;
+    const i18nErrors = [];
+    const warnings = [];
+    const errorArgumentsRegex = /{([^}]+)}/g;
+
+    for (const err of installabilityErrors) {
+      // Filter out errorId 'in-incognito' since Lighthouse recommends incognito.
+      if (err.errorId === 'in-incognito') continue;
+
+      if (err.errorId === 'warn-not-offline-capable') {
+        warnings.push(str_(UIStrings[err.errorId]));
+        continue;
+      }
+
+      // @ts-expect-error errorIds from protocol should match up against the strings dict
+      const matchingString = UIStrings[err.errorId];
+
+      if (matchingString === undefined) {
+        i18nErrors.push(str_(UIStrings.noErrorId, {errorId: err.errorId}));
+        continue;
+      }
+
+      // Get the arguments of the installability error message.
+      const UIStringArguments = matchingString.match(errorArgumentsRegex) || [];
+
+      /**
+       * If there is an argument value, get it.
+       * We only expect a `minimum-icon-size-in-pixels` errorArg[0] for two errorIds, currently.
+       */
+      const value0 = err.errorArguments && err.errorArguments.length && err.errorArguments[0].value;
+
+      if (matchingString && err.errorArguments.length !== UIStringArguments.length) {
+        // Matching string, but have the incorrect number of arguments for the message.
+        const argsReceived = err.errorArguments.join(', ');
+        const msg = err.errorArguments.length > UIStringArguments.length ?
+          ` has unexpected arguments {${argsReceived}}` :
+          ` does not have the expected number of arguments.`;
+        i18nErrors.push(msg);
```

Comment:
should we be putting this in `warnings` or something more prominent than inline with the other errors? I'm not sure what's a good place where we'll ever actually see it

Reply:
iirc devtools has them inline with their errors, where this msg would act as the error message. But, I don't mind having them as warnings, since the situation in which they come up is a little different than a regular error message. Although, if we put it in warnings they won't be shown on the report 👀

Reply:
Hmm, maybe we're talking about different things? I meant the audit `warnings` array

Reply:
Also, let's just ignore this `warnings` chain of thought. Sentry is another option. If it's a problem after launch we can address it then (after all, we're hoping this case just never happens as long as we stay reasonably up to date with the protocol)